Neos.Neos/Classes/Domain/NodeLabel/ExpressionBasedNodeLabelGenerator.php

```diff
@@ -63,9 +61,37 @@ public function setExpression(string $expression): void
      */
     public function getLabel(Node $node): string
     {
-        if (Utility::parseEelExpression($this->getExpression()) === null) {
-            return $this->getExpression();
+        $contentRepository = $this->contentRepositoryRegistry->get($node->contentRepositoryId);
+        $nodeTypeManager = $contentRepository->getNodeTypeManager();
+        $nodeTypeManager->getNodeType($node->nodeTypeName);
+        $subgraph = $this->contentRepositoryRegistry->subgraphForNode($node);
+
+        $expression = $this->getExpression();
+        $parentNode = $subgraph->findParentNode($node->aggregateId);
+        $parentNodeType = $nodeTypeManager->getNodeType($parentNode?->nodeTypeName);
```

Comment:
This will crash, if the $parentNode is null.
```suggestion
        $parentNodeType = $parentNode ? $nodeTypeManager->getNodeType($parentNode->nodeTypeName) : null;
```